distribution/docs/src/main/jdocs/content/_developingComponents/developing-plugins-contents.adoc

```diff
@@ -0,0 +1,390 @@
+:title:  Developing Catalog Plugins
+:type: developingComponent
+:status: published
+:link: _developing_catalog_plugins
+:summary: Creating a custom catalog plugin.
+
+Plugins extend the functionality of the Catalog Framework by performing actions at specified times during a transaction. 
+Plugin interfaces are located in the Catalog Core API.
+By implementing a plugin interface, actions can be performed at the desired time. 
+
+[NOTE]
+====
+Catalog plugin interfaces are located in the catalog core API.
+====
+
+The following types of plugins can be created:
+
+.Plugin Interfaces
+[cols="1,2m,1,1", options="header"]
+|===
+
+|Plugin Type
+|Plugin Interface
+|Description
+|Example
+
+|Pre-Ingest
+|ddf.catalog.plugin.PreIngestPlugin
+|Runs before the Create/Update/Delete method is sent to the CatalogProvider
+|Metadata validation services
+
+|Post-Ingest
+|ddf.catalog.plugin.PostIngestPlugin
+|Runs after the Create/Update/Delete method is sent to the CatalogProvider
+|EventProcessor for processing and publishing event notifications to subscribers
+
+|Pre-Query
+|ddf.catalog.plugin.PreQueryPlugin
+|Runs prior to the Query/Read method being sent to the Source
+|An example is not included with ${branding}
+
+|Post-Query
+|ddf.catalog.plugin.PostQueryPlugin
+|Runs after results have been retrieved from the query but before they are posted to the Endpoint
+|An example is not included with ${branding}
+
+|Pre-Subscription
+|ddf.catalog.plugin.PreSubscription
+|Runs prior to a Subscription being created or updated
+|Modify a query prior to creating a subscription
+
+|Pre-Delivery
+|ddf.catalog.plugin.PreDeliveryPlugin
+|Runs prior to the delivery of a Metacard when an event is posted
+|Inspect a metacard prior to delivering it to the Event Consumer
+
+|Pre-Resource
+|ddf.catalog.plugin.PreResource
+|Runs prior to a Resource being retrieved
+|An example is not included with ${branding}
+
+|Post-Resource
+|ddf.catalog.plugin.PostResource
+|Runs after a Resource is retrieved, but before it is sent to the Endpoint
+|Verification of a resource prior to returning to a client
+
+|Policy
+|ddf.catalog.plugin.PolicyPlugin
+|Runs prior to all other catalog plugins to esablish the policy for requests/responses
```

Comment:
Actually, that's a good catch. I don't think the policy plugins run first anymore. 
I purposely didn't add an `order` property to list the plugins in order but I could add one if it adds any value to navigating through them. I'm not sure if people would be looking for a plugin by name or by type more often. If it's by name, alphabetical order is probably easiest. It's easy enough to change.